### PR TITLE
[WIP] Use real hg pushlog / fig -> docker-compose

### PR DESCRIPTION
**Pull request by lightsofapollo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@rail This is fairly small test only set of changes ... I had hoped to add the core of the hg pushlog refactor here but this bit at least does some useful refactoring of how docker-compose is downloaded/used. If you want to give this a skim it probably is good starting point for asking "wtf is ..." type questions...

File: test/compose.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+@param {String} [installPath] install path for compose binaries.
+*/
+export async function install(installPath = COMPOSE_INSTALL_ROOT) {
+  let { url, slug } = composeLocationConfig();
+  debug('compose install', { url, slug });
+
+  // Ensure compose install root exists...
+  if (!await fs.exists(installPath)) {
+    await fs.mkdir(installPath);
```
Comment (rail) [inline test/compose.js]: There is a very little chance to get a race condition between fs.exists() and fs.mkdir(). I'd rather call fs.mkdir() and ignore EEXIST.

Also would be better to set mode to something safer than default 777.

File: test/compose.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  let composeBin = fsPath.join(installPath, slug);
+
+  // Check for an existing install ..
+  if (await fs.exists(composeBin)) {
+    debug('compose already installed...');
+    return new Compose(composeBin);
+  }
```
Comment (rail) [inline test/compose.js]: I think this block can be moved before the one with fs.mkdir().

**Comment by rail** [comment]
In overall LGTM.

File: test/compose.js (hunk 2 above)
Comment (lightsofapollo) [inline test/compose.js]: the install path may exist but the slug may not (in case of upgrading versions)

Comment (rail) [inline test/compose.js]: ah, I see

**Pull request merged**